### ISSUE TITLE
Enhance the deploy command to enable the setting of git user.name and user.email in deployment branch

### ISSUE THREAD

**Issue by amdshrif** [facebook/docusaurus#9101]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Description

Enhance the deploy command to enable the setting of git user.name and user.email specifically for the deployment branch. This way, Git will utilize the local git configuration instead of the global configuration when making commits to the deployment branch.

### Has this been requested on Canny?

NA

### Motivation

In my system, I have separate configurations for my personal and work Git users. These configurations are set on a per-repository basis rather than globally. However, when I use the command "yarn deploy," it automatically sets the user.name and user.email to default values, which are based on my hostname. I would rather not share this information on GitHub and have the option to set the user.name and user.email that I would like to use for commiting changes to deployment branch.

### API design

I have added 2 environment variables to set the user.name and user.email  

### Have you tried building it?

Yes, I have added 2 environment variables to set the user.name and user.email  

### Self-service

- [X] I'd be willing to contribute this feature to Docusaurus myself.

**Comment by slorber** [comment]
I don't think we need a change, see my comment here: https://github.com/facebook/docusaurus/pull/9102#pullrequestreview-1505003048

> However, when I use the command "yarn deploy," it automatically sets the user.name and user.email to default values, which are based on my hostname.

It's your responsibility to ensure that your git config is good before running that CLI.

We currently don't "automatically sets" any config in the CLI, and I don't think we should.
